source/containers/DynamicCardRenderer/DynamicCardRenderer.tsx

```diff
@@ -156,62 +171,46 @@ const renderCardComponent = (
           circle={component.circle}
         />
       );
-  }
-
-  // Treat buttons separately, because they have some more complicated behavior
-  if (component.type === "button") {
-    const { icon, iconPosition, text } = component;
-    const onClick: () => void = () => null;
+    case "button":
+      if (component.action === "infoModal") {
+        return (
+          <InfoModalButton key={`${index}-${component.type}`} {...component} />
+        );
+      }
 
-    // treat info-modal separately since it doesn't fit the same pattern as the other buttons.
-    if (component.action === "infoModal") {
       return (
-        <InfoModalButton key={`${index}-${component.type}`} {...component} />
+        <Card.Button
+          key={`${index}-${component.type}`}
+          onClick={handleClick(component, navigation)}
+        >
+          {component.icon &&
+            component.iconPosition &&
+            component.iconPosition === "left" && <Icon name={component.icon} />}
+          <TextComponent>{component.text}</TextComponent>
+          {component.icon &&
+            (!component.iconPosition || component.iconPosition === "right") && (
+              <Icon name={component.icon} />
+            )}
+        </Card.Button>
       );
```

Comment:
The return of Card.Button and its icons in this case is quite verbose. You could do a small refactoring like below to make it a little more readable:
_Notice the logic for deciding on which icon position to render_

```js
    case "button": {
      const { iconPosition = "right", icon } = component;
      if (component.action === "infoModal") {
        return (
          <InfoModalButton key={`${index}-${component.type}`} {...component} />
        );
      }

      return (
        <Card.Button
          key={`${index}-${component.type}`}
          onClick={handleClick(component, navigation)}
        >
          {icon && iconPosition === "left" && <Icon name={icon} />}

          <TextComponent>{component.text}</TextComponent>

          {icon && iconPosition === "right" && <Icon name={icon} />}
        </Card.Button>
      );
    }
```

Reply:
Done!